web/src/compose_recipient.js

```diff
@@ -97,13 +100,47 @@ export function open_compose_stream_dropup() {
     $("#id_compose_select_recipient > .dropdown-toggle").dropdown("toggle");
 }
 
+function switch_message_type(message_type) {
+    $("#compose-content .alert").hide();
+
+    compose_state.set_message_type(message_type);
+
+    const opts = {
+        message_type,
+        trigger: "toggle message",
+        stream: compose_state.stream_name(),
+        topic: compose_state.topic(),
+        private_message_recipient: compose_state.private_message_recipient(),
+    };
+    compose_actions.show_compose_box(message_type, opts);
+}
+
 export function on_compose_select_recipient_update(new_value) {
-    const $stream_header_colorblock = $("#compose_recipient_selection_dropdown").find(
-        ".stream_header_colorblock",
-    );
-    stream_bar.decorate(new_value, $stream_header_colorblock);
+    const message_type = compose_state.get_message_type();
+    if (new_value === PRIVATE_MESSAGE) {
+        // TODO: maybe make a way for the dropdown to have a different
+        // string show on the button than in the list.
+        $("#compose_select_recipient_name").text($t({defaultMessage: "PM"}));
+        if (message_type !== "private") {
+            switch_message_type("private");
+        }
+        if (compose_state.private_message_recipient().length === 0) {
```

Comment:
I assume this is only going to be non-empty if we were already there? Ahh, but it could have been empty before opening the input and still empty after even though we re-picked "DMs", so moving this into the `message_type !== "private"` block would not be correct.

Reply:
We keep the value even when switching to stream, so yes it can be non-empty when switching back.